File: docs/julia/algorithmes/cmd_args_file_io.jl
#  First programs in Julia.
#  Julien VALENTIN (2022)  julien.vlnt@gmail.com
#  Compute the sum of the two first arguments and write the result in a file.
#  usage : `julia computer.jl <m> <n> <ouputFile>`

# if length(ARGS) != 3 : exit.
if length(ARGS) != 3
    println("usage : julia computer.jl <m> <n> -o <output txt file>")
else
    # Arguments are represented under the type String.
    m = parse(Int64, ARGS[1])
    n = parse(Int64, ARGS[2])
    
    # Compute the sum.
    var = m + n
    
    # Write the result in the file.
    try
        open(ARGS[3], "w") do f
            write(f, "$(var)\n")
        end
    catch
        println("Error writing in the file !")
    end

    # Sleep five seconds...
    sleep(5)

    # ... then delete the file.
    rm(ARGS[3])
end

File: docs/c/algorithmes/cmd_args_file_io.c
//  First programs in C
//  Julien VALENTIN (2022)  julien.vlnt@gmail.com
//  Compute the sum of the two first arguments and write the result in a file.
//  Usage : `computer.exe <m> <n> <ouputFile>`.
#include <stdio.h>
#include <Windows.h>

int main(int argc, char *argv[])
{
    int m
    ,   n
    ,   var;

    // If argc != 4 : exit.
    if(argc != 4)
    {
        printf("Usage : `computer.exe <m> <n> <outputfile>\n");
    }

    else
    {
        // Translate numbers into integers.
        m = atoi(argv[1]);
        n = atoi(argv[2]);
        
        // Compute the sum.
        var = m + n;

        // Try to open the file.
        FILE *f = fopen(argv[3], "w");

        if(f != NULL)    // opening in "writing" mode is a success.
        {
            // write the result
            fprintf(f, "%i", var);
            
            // close the file before the program ends.
            fclose(f);

            // wait for five seconds...
            Sleep(5000);

            // ... then delete the file.
            remove(argv[3]);
        }
        else
        {
            // Error while opening the file.
            printf("Error opening file <%s>. Exit !", argv[3]);
        }
    
    }

    return 0;
}